corehq/apps/user_importer/helpers.py

```diff
@@ -9,3 +16,269 @@ def spec_value_to_boolean_or_none(user_spec_dict, key):
+        # ToDo: consider putting just the diff
+        if self.logger.original_user_doc['user_data'] != self.user.user_data:
+            self.logger.fields_changed['user_data'] = self.user.user_data
+            self._save = True
+
+
+class CommCareUserImporter(BaseUserImporter):
+    def update_password(self, password):
+        self.user.set_password(password)
+        self.logger.add_change_message(_("Password Reset"))
+
+    def update_phone_number(self, phone_number):
+        fmt_phone_number = _fmt_phone(phone_number)
+        if fmt_phone_number not in self.user.phone_numbers:
+            self.logger.add_change_message(_("Added phone number {new_phone_number}").format(
+                new_phone_number=fmt_phone_number)
+            )
+        # always call this to set phone number as default if needed
+        self.user.add_phone_number(fmt_phone_number, default=True)
+
+    def update_name(self, name):
+        self.user.set_full_name(str(name))
+        self.logger.add_changes({'first_name': self.user.first_name, 'last_name': self.user.last_name})
+
+    def update_user_data(self, data, uncategorized_data, profile, domain_info):
+        # Add in existing data. Don't use metadata - we don't want to add profile-controlled fields.
+        for key, value in self.user.user_data.items():
+            if key not in data:
+                data[key] = value
+        if profile:
+            profile_obj = domain_info.profiles_by_name[profile]
+            data[PROFILE_SLUG] = profile_obj.id
+            for key in profile_obj.fields.keys():
+                self.user.pop_metadata(key)
+        try:
+            self.user.update_metadata(data)
+        except ValueError as e:
+            raise UserUploadError(str(e))
+        if uncategorized_data:
+            self.user.update_metadata(uncategorized_data)
+
+        # Clear blank user data so that it can be purged by remove_unused_custom_fields_from_users_task
+        for key in dict(data, **uncategorized_data):
+            value = self.user.metadata[key]
+            if value is None or value == '':
+                self.user.pop_metadata(key)
+
+    def update_language(self, language):
+        self.user.language = language
+        self.logger.add_changes({'language': language})
+
+    def update_email(self, email):
+        self.user.email = email.lower()
+        self.logger.add_changes({'email': self.user.email})
+
+    def update_status(self, is_active):
+        self.user.is_active = is_active
+        self.logger.add_changes({'is_active': is_active})
+
+    def update_locations(self, location_codes, domain_info):
+        from corehq.apps.user_importer.importer import (
+            check_modified_user_loc,
+            find_location_id,
+            get_location_from_site_code
+        )
+
+        location_ids = find_location_id(location_codes, domain_info.location_cache)
+        users_current_primary_location_id = self.user.location_id
```

Comment:
nitpick: `user_current_primary_location_id` instead of `users_current_primary_location_id`

Reply:
thank you. 
Updated this and one more
https://github.com/dimagi/commcare-hq/pull/29935/commits/d3898ed2ae37199cdacd96bbafc405d7621dc4e3